apps/client/src/routes/dashboards/dashboard/hooks/use-dashboard-query.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from '@tanstack/react-query';
+import invariant from 'tiny-invariant';
+
+import { createDashboardQuery } from '~/data/dashboards';
+import { isFatal } from '~/helpers/predicates/is-fatal';
+import { isNotFatal } from '~/helpers/predicates/is-not-fatal';
+import { useEmitNotification } from '~/hooks/notifications/use-emit-notification';
+import { GenericErrorNotification } from '~/structures/notifications/generic-error-notification';
+
+import type { Dashboard } from '~/services';
+
+export function useDashboardQuery(dashboardId: Dashboard['id']) {
+  const emit = useEmitNotification();
+
+  return useQuery({
+    ...createDashboardQuery(dashboardId),
+    useErrorBoundary: isFatal,
+    onError: (error) => {
+      invariant(
+        isNotFatal(error),
+        'Expected fatal error to be handled by error boundary',
+      );
```

Comment:
Might not be a good usage of `invariant(...)` here. During fatal error, the error boundary is getting `Error('Invariant violation: ...` and the actual `error` is not passed.